File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
    let left = 0
    let right = numbers.length - 1;
    while (left < right) {
        const currentValue = numbers[left] + numbers[right];
        if (currentValue > target) {
            right--;
        } else if (target > currentValue) {
            left++;
        } else {
            return [left+1, right+1];
        }
    }

    // Using Hash Map
    // Problem is that the space used is O(n) pointers will be O(1)
    // const numberSet = new Map()
    // for(let i = 0; i < numbers.length; i++ ){
    //     if (numberSet.has(target - numbers[i])){
    //         return [numberSet.get(target - numbers[i]), i+1];
    //     }
    //     numberSet.set(numbers[i], i + 1);
    // }
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.rb
# @param {Integer[]} numbers
# @param {Integer} target
# @return {Integer[]}
def two_sum(numbers, target)
    left_pointer_index = 0;
    right_pointer_index = numbers.length()-1;
    while left_pointer_index < right_pointer_index
        if numbers[left_pointer_index] + numbers[right_pointer_index] > target
            right_pointer_index -= 1
        elsif numbers[left_pointer_index] + numbers[right_pointer_index] < target
            left_pointer_index += 1
        else
            return [left_pointer_index+1, right_pointer_index+1]   
        end
    end
end